### DISCUSSION TITLE
Validation between elements of BusinessBindingListBase

### DISCUSSION THREAD

**Discussion by pszittyay** [MarimerLLC/csla#1829] [Questions] [answered]
Hi Folks,

I have a BusinessBindingListBase and i want to validate that the elements of the list between each other. I must ensure that some values do not overlap between them. 
I have already added a BusinessRule in the child class following the example https://github.com/MarimerLLC/csla/discussions/1795 and it works fine, validating against the values already in the db. The problem comes when the users add multiples elements that violate the rule between them and not againts the db. The final results is that the elements get inserted and the expected result is a validation error.

**Comment by rockfordlhotka** [comment, marked as answer]
There are two scenarios:

1. You have a BusinessBase parent for the list
1. The list is the root of the graph

For option 1 the easiest thing is to implement a rule that checks the list for conflicts, then trigger that rule in the BusinessBase parent by handling the OnChildChanged or ChildChanged event.

For option 2 there's no rules engine in list objects, but you can handle the OnChildChanged/ChildChanged event and implement logic to check the list for conflicts. The trick here, is that only the child objects can have any visual aspect (the list itself isn't "visible" via data bnding). So the logic in your list class will need to call something in one or more child objects so those child objects become invalid.
